sycl/test/basic_tests/ordered_buffs.cpp

```diff
@@ -0,0 +1,85 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//==-------- ordered_buffs.cpp - SYCL buffers in ordered queues test--------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+const int dataSize = 32;
+
+bool getQueueOrder(cl_command_queue cq) {
+  cl_command_queue_properties reportedProps;
+  cl_int iRet = clGetCommandQueueInfo(
+      cq, CL_QUEUE_PROPERTIES, sizeof(reportedProps), &reportedProps, nullptr);
+  assert(CL_SUCCESS == iRet && "Failed to obtain queue info from ocl device");
+  return (reportedProps & CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE) ? false
+                                                                  : true;
+}
+
+int main() {
+  int data[dataSize] = {0};
+
+  {
+    ordered_queue Queue;
+
+    // Purpose of this test is to create a dependency between two kernels
+    // RAW dependency
+    // which requires the use of ordered queue.
+    buffer<int, 1> bufA(data, range<1>(dataSize));
```

Comment:
It's illegal to create two sycl buffers from the same host ptr.

Reply:
I could not find this restriction in the spec. Can you please point me to it? Also if this is illegal, how come we don't encounter issues at compile or runtime?

Reply:
Let's address that separately.  For now can we just have dataA and data B?

Reply:
OK